icu4c/source/test/intltest/dtifmtts.cpp

```diff
@@ -1806,6 +1807,97 @@ void DateIntervalFormatTest::testCreateInstanceForAllLocales() {
     }
 }
 
+void DateIntervalFormatTest::testFormatMillisecond() {
+    struct
+    {
+        int year;
+        int month;
+        int day;
+        int from_hour;
+        int from_miniute;
+        int from_second;
+        int from_millisecond;
+        int to_hour;
+        int to_miniute;
+        int to_second;
+        int to_millisecond;
+        const char* skeleton;
+        const char* expected;
```

Comment:
Optional: it's nicer to make this `const char16_t*`, use `u""` strings, and then use UnicodeString instead of std::string, and you can also use `\u2013` directly (even write it literally into the file).